lib/libthr/arch/aarch64/include/pthread_md.h

```diff
@@ -57,7 +57,11 @@ static __inline void
 _tcb_set(struct tcb *tcb)
 {
 
+#ifdef __CHERI_PURE_CAPABILITY__
+	__asm __volatile("msr	ctpidr_el0, %x0" :: "C" (tcb));
```

Comment:
What does the x modifier do and why is it here but not in `_tcb_get` for purecap?

Reply:
Ok, reading the LLVM (and rather harder-to-follow GCC) source, it's basically a way to ensure you get an X register even if the operand gets allocated to a 32-bit W sub-register. Since we use `C` that doesn't make any sense (and to be honest this should be an error IMO, or emit the X register and fail to assemble) and this should just be `%0`, but it happens to work because LLVM defaults to using the original allocated register if it wasn't a W register.

(There's also a corresponding `w` modifier for X-to-W that is also similarly "broken", i.e. does nothing, when given a capability)